### PR TITLE
[ticket/16935] Refactor sphinx to use new clases and remove unused parts

### PR DESCRIPTION
**Pull request by marc1706**
PHPBB3-16935

Checklist:

- [x] Correct branch: master for new features; 3.3.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/master/coding-guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-16935

### CODE REVIEW

File: phpBB/phpbb/search/backend/fulltext_sphinx.php [hunk 1]
```diff
@@ -631,7 +631,7 @@ public function tidy(): void
 	 */
 	public function create_index(int &$post_counter = 0): ?array
 	{
-		if ($this->index_created())
```
Comment (rubencm) [inline phpBB/phpbb/search/backend/fulltext_sphinx.php:634]: I am not sure why I did this.

Comment (rubencm) [reply phpBB/phpbb/search/backend/fulltext_sphinx.php:634]: Probably the truncate table lines can be removed

Comment (marc1706) [reply phpBB/phpbb/search/backend/fulltext_sphinx.php:634]: Yeah, truncating a newly created table sounds kind of wrong.

File: tests/functional/search/sphinx_test.php [hunk 2]
```diff
@@ -20,8 +20,29 @@ class phpbb_functional_search_sphinx_test extends phpbb_functional_search_base
...
+			exec('sudo -S service sphinxsearch stop', $output, $retval); // Attemtp to stop sphinxsearch service in case it's running
+			exec('sudo -S indexer --all', $output, $retval); // Run sphinxsearch indexer
+			exec('sudo -S service sphinxsearch start', $output, $retval); // Attempt to start sphinxsearch service again
+		}
+	}
+
 	public function test_search_backend()
 	{
-		$this->markTestIncomplete('Sphinx Tests are not supported');
+		if ($this->db->sql_layer != 'mysqli') // Sphinx test runs on MySQL/MariaDB only so far
```
Comment (rubencm) [inline tests/functional/search/sphinx_test.php:39]: https://github.com/phpbb/phpbb/blob/master/phpBB/phpbb/search/backend/fulltext_sphinx.php#L203

Comment (marc1706) [reply tests/functional/search/sphinx_test.php:39]: Generally yes, but this is with regards to the unit testing. The sample file is mysql only and the same applies for the tests on GitHub Actions. Therefore I would keep it like this for now. Created a ticket here: https://tracker.phpbb.com/browse/PHPBB3-16954

**Pull request merged**